rust/src/countme.rs

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (C) 2020 Red Hat, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+use anyhow::{bail, Context, Result};
+use curl::easy::Easy;
+use os_release::OsRelease;
+use std::path;
+
+mod cookie;
+mod repo;
+
+/// Default variant name used in User Agent
+const DEFAULT_VARIANT_ID: &str = "unknown";
+
+/// Send a request to 'url' with 'ua' as User Agent.
+/// This sends a GET request and discards the body as this is what is currently
+/// expected on the Fedora infrastructure side.
+/// Once this is fixed, we can switch to a HEAD request to reduce the footprint:
+/// let mut handle = Easy::new().nobody(true)?;
+fn send_countme(url: &str, ua: &str) -> Result<()> {
+    println!("Sending request to: {}", url);
+    let mut handle = Easy::new();
```

Comment:
Let's share the basic setup here with the curl bits in `utils.rs` - notably I think we want
```
    handle.follow_location(true)?;
    handle.fail_on_error(true)?;
```
too.